src/lib/utils.ts

```diff
@@ -10,3 +11,79 @@ export const baseURL = process.env.NEXT_PUBLIC_URL
   : process.env.NEXT_PUBLIC_VERCEL_URL
   ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}/`
   : `http://localhost:${process.env.PORT ?? 3000}/`;
+
+declare global {
+  interface Window {
+    Razorpay: any;
+  }
+}
+
+export interface makePaymentProps {
+  productId: string | null;
+  productName: string;
+  description: string;
+  amount: number;
+  prefillData: {
+    name: string;
+    email: string;
+    contact: string;
+  };
+}
+
+export const makePayment = async ({
+  productId = null,
+  productName,
+  description,
+  prefillData,
+  amount,
+}: makePaymentProps) => {
+  const key = process.env.RAZORPAY_API_KEY;
+  const response = await fetch("/api/razorpay", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json", // Assuming the API expects JSON
+    },
+    body: JSON.stringify({ amount }), // Pass amount as a parameter in the request body
+  });
+  const { orderId } = await response.json();
+  const options = {
+    key: key,
+    name: productName,
+    currency: "INR",
+    amount: amount,
+    order_id: orderId,
+    description: description,
+    prefill: {
+      name: prefillData.name,
+      email: prefillData.email,
+      contact: prefillData.contact,
+    },
+    handler: async function (response: any) {
+      const data = {
+        orderCreationId: orderId,
+        razorpayPaymentId: response.razorpay_payment_id,
+        razorpayOrderId: response.razorpay_order_id,
+        razorpaySignature: response.razorpay_signature,
+       };
+  
+       const result = await fetch('/api/razorpay/verify', {
+        method: 'POST',
+        body: JSON.stringify(data),
+        headers: { 'Content-Type': 'application/json' },
+       });
+       const res = await result.json();
+       if (res.isOk) alert("payment succeed");
+       else {
+        alert(res.message);
+       }
+    },
+  };
+  
+  const paymentObject = new window.Razorpay(options);
+  
+  paymentObject.on("payment.failed", function (response: any) {
+    alert("Payment failed. Please try again. Contact support for help"+response.error.description);
```

Comment:
Ensure you use, `toast` or `Sonner` before pushing this

Reply:
yup